pkgs/applications/editors/mindforger/default.nix

```diff
@@ -1,37 +1,56 @@
 { lib
 , stdenv
-, cmark-gfm
-, fetchurl
-, fetchpatch
+, cmake
+, curl
+, fetchFromGitHub
+, hunspell
+, pkg-config
 , qmake
 , qtbase
 , qtwebengine
 , wrapGAppsHook3
 , wrapQtAppsHook
 }:
-
-stdenv.mkDerivation rec {
+let
+  version = "2.0.0";
+  srcs = {
+    mindforger = fetchFromGitHub {
+      owner = "dvorka";
+      repo = "mindforger";
+      rev = version;
+      sha256 = "sha256-+8miV2xuQcaWGdWCEXPIg6EXjAHtgD9pX7Z8ZNhpMjA=";
+    };
+    cmark-gfm = fetchFromGitHub {
+      owner = "dvorka";
+      repo = "cmark";
+      rev = "4ca8688093228c599432a87d7bd945804c573d51";
+      sha256 = "sha256-0WiG8aot8mc0h1BKPgC924UKQrgunZvKKBy9bD7nhoQ=";
+    };
+  };
+in stdenv.mkDerivation {
   pname = "mindforger";
-  version = "1.52.0";
+  inherit version;
 
-  src = fetchurl {
-    url = "https://github.com/dvorka/mindforger/releases/download/${version}/mindforger_${version}.tgz";
-    sha256 = "1pghsw8kwvjhg3jpmjs0n892h2l0pm0cs6ymi8b23fwk0kfj67rd";
-  };
+  src = srcs.mindforger;
+
+  nativeBuildInputs = [ cmake pkg-config qmake wrapGAppsHook3 wrapQtAppsHook ];
+  buildInputs = [ curl hunspell qtbase qtwebengine ];
 
-  nativeBuildInputs = [ qmake wrapGAppsHook3 wrapQtAppsHook ];
-  buildInputs = [ qtbase qtwebengine cmark-gfm ];
+  # Disable the cmake hook (so we don't try to build MindForger with it), and
+  # build MindForger's internal fork of cmark-gfm ahead of MindForger itself:
+  postUnpack = "cp -TR ${srcs.cmark-gfm} $sourceRoot/deps/cmark-gfm";
+  dontUseCmakeConfigure = true;
+  preBuild = ''(
+      mkdir deps/cmark-gfm/build &&
+      cd deps/cmark-gfm/build &&
+      cmake -DCMARK_TESTS=OFF -DCMARK_SHARED=OFF .. &&
+      cmake --build . --parallel
+  )'';
 
   doCheck = true;
 
   patches = [
```

Comment:
Please set meta.platforms = linux or fix
```
       > ranlib -s libmindforger.a
       > ranlib: file: libmindforger.a(link.o) has no symbols
       > ranlib: file: libmindforger.a(naive_autolinking_preprocessor.o) has no symbols
       > make[1]: Leaving directory '/private/tmp/nix-build-mindforger-2.0.0.drv-0/source/lib'
       > cd app/ && ( test -e Makefile || /nix/store/n84f7j30ni38zl56rfih32nc6yp30qcd-qtbase-5.15.12-dev/bin/qmake -o Makefile /private/tmp/nix-build-mindforger-2.0.0.drv-0/source/app/app.pro PREFIX=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0 NIX_OUTPUT_OUT=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0 NIX_OUTPUT_DEV=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0 NIX_OUTPUT_BIN=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0 NIX_OUTPUT_DOC=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0/share/doc/qt-5.15.12 NIX_OUTPUT_QML=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0/lib/qt-5.15.12/qml NIX_OUTPUT_PLUGIN=/nix/store/ma3qya28jjrwpbq04pasw9yl1pw9p2j1-mindforger-2.0.0/lib/qt-5.15.12/plugins CONFIG+=release CONFIG+=nostrip CONFIG+=mfnoccache CONFIG+=mfwebengine ) && make -f Makefile
       > make[1]: Entering directory '/private/tmp/nix-build-mindforger-2.0.0.drv-0/source/app'
       > make[1]: *** No rule to make target '../doc/limbo', needed by 'mindforger.app/Contents/Resources/mindforger-repository/limbo'.  Stop.
       > make[1]: Leaving directory '/private/tmp/nix-build-mindforger-2.0.0.drv-0/source/app'
       > make: *** [Makefile:76: sub-app-make_first] Error 2
```

Reply:
On which platform are you getting that? Darwin?

Reply:
What seems to happen is that the Mindforger packages the documentation on MacOS [1], and it's pulled to the doc folder as a git submodule [2].

[1]: https://github.com/dvorka/mindforger/blob/de1da1d50582d3a34aa977bbd689505c3992a669/app/app.pro#L494
[2] https://github.com/dvorka/mindforger/blob/de1da1d50582d3a34aa977bbd689505c3992a669/.gitmodules#L1

Would mind trying something like:

```diff
diff --git a/pkgs/applications/editors/mindforger/default.nix b/pkgs/applications/editors/mindforger/default.nix
index 722a425ca27b..bbda7a063c77 100644
--- a/pkgs/applications/editors/mindforger/default.nix
+++ b/pkgs/applications/editors/mindforger/default.nix
@@ -26,6 +26,12 @@ let
       rev = "4ca8688093228c599432a87d7bd945804c573d51";
       sha256 = "sha256-0WiG8aot8mc0h1BKPgC924UKQrgunZvKKBy9bD7nhoQ=";
     };
+    mindforger-repository = {
+      owner = "dvorka";
+      repo = "mindforger-repository";
+      rev = "4ca8688093228c599432a87d7bd945804c573d51";
+      sha256 = "";
+    };
   };
 in stdenv.mkDerivation {
   pname = "mindforger";
@@ -37,8 +43,13 @@ in stdenv.mkDerivation {
   buildInputs = [ curl hunspell qtbase qtwebengine ];

   # Disable the cmake hook (so we don't try to build MindForger with it), and
-  # build MindForger's internal fork of cmark-gfm ahead of MindForger itself:
-  postUnpack = "cp -TR ${srcs.cmark-gfm} $sourceRoot/deps/cmark-gfm";
+  # build MindForger's internal fork of cmark-gfm ahead of MindForger itself.
+  #
+  # Moreover unpack the docs that are needed for the MacOS build.
+  postUnpack = ''
+    cp -TR ${srcs.cmark-gfm} $sourceRoot/deps/cmark-gfm
+    cp -TR ${srcs.mindforger-repository} $sourceRoot/doc
+  '';
   dontUseCmakeConfigure = true;
   preBuild = ''(
       mkdir deps/cmark-gfm/build &&
```

On top of this PR?

Reply:
Well I can't. I just looked at ofborg's build attempt. (You can push your fix with the right hash to test)